sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -201,6 +215,15 @@ public WritableColumnVector reserveDictionaryIds(int capacity) {
    */
   public abstract void putBooleans(int rowId, int count, boolean value);
 
+  /**
+   * Sets bits from [src[srcIndex], src[srcIndex + count]) to [rowId, rowId + count)
+   * src must be positive and contain 8 bits of bitmask in the lowest byte.
+   */
+  public void putBooleans(int rowId, int count, int src, int srcIndex) {
+    putBytes(rowId, count, ByteBuffer.wrap(byte8).putLong(toBitPerByte(src)).array(), srcIndex);
+  }
+  public void putBooleans(int rowId, int src) { putBooleans(rowId, 8, src, 0); }
```

Comment:
style nit: let's start a new line for the method body.

Reply:
Updated